src/main/java/edu/wpi/grip/ui/preview/ContoursSocketPreviewView.java

```diff
@@ -0,0 +1,84 @@
+package edu.wpi.grip.ui.preview;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import edu.wpi.grip.ui.util.ImageConverter;
+import javafx.application.Platform;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.VBox;
+
+import static org.bytedeco.javacpp.opencv_core.*;
+import static org.bytedeco.javacpp.opencv_imgproc.drawContours;
+import static edu.wpi.grip.core.operations.composite.FindContourOperation.Contours;
+
+/**
+ * A preview view for displaying contours.  This view shows each contour as a different-colored outline (so they can be
+ * individually distinguished), as well as a count of the total number of contours found.
+ */
+public class ContoursSocketPreviewView extends SocketPreviewView<Contours> {
+
+    private final ImageConverter imageConverter = new ImageConverter();
+    private final ImageView imageView = new ImageView();
+    private final Label infoLabel = new Label();
+    private final Mat tmp = new Mat();
+
+    private final static Scalar[] CONTOUR_COLORS = new Scalar[]{
+            Scalar.RED,
+            Scalar.YELLOW,
+            Scalar.GREEN,
+            Scalar.CYAN,
+            Scalar.BLUE,
+            Scalar.MAGENTA,
+    };
+
+    /**
+     * @param eventBus The EventBus used by the application
+     * @param socket   An output socket to preview
+     */
+    public ContoursSocketPreviewView(EventBus eventBus, OutputSocket<Contours> socket) {
+        super(eventBus, socket);
+
+        this.setContent(new VBox(this.imageView, this.infoLabel));
+        this.render();
```

Comment:
Again, we talked about this a while ago.  We're doing this for now because it probably won't actually cause any problems, but in the long term we're going to use dependency injection